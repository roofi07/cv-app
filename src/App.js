import React, { Component } from 'react'
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect,
} from 'react-router-dom'

import './App.scss'
import Creator from './pages/Creator/Creator'

class App extends Component {
  render() {
    return (
      <Router basename="">
        <div className="container">
          <Switch>
            <Route component={Creator} exact path="/creator" />
            <Redirect to="/creator" />
          </Switch>
        </div>
      </Router>
    )
  }
}

export default App
