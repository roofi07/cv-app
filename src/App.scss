body {
  background-color: #ccc;
}

* {
  box-sizing: border-box;
}
