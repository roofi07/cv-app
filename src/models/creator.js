import Immutable from 'immutable'
import window from '../utils/window'

const creator = {
  effects: {
    async getCreator() {
      const { localStorage } = window
      try {
        const jsonData = localStorage.getItem('CV')
        const modulesData = JSON.parse(jsonData) || []
        this.update(modulesData)
      } catch (err) {
        throw new Error(err)
      }
    },
    async setCreator(data) {
      const { localStorage } = window
      try {
        const jsonData = JSON.stringify(data.toJS())
        await localStorage.setItem('CV', jsonData)
        this.update(data)
      } catch (err) {
        throw new Error(err)
      }
    },
  },
  reducers: {
    update(state, payload) {
      const modulesData = Immutable.fromJS(payload)
      return { ...state, modulesData }
    },
  },
  state: {
    modulesData: Immutable.List(),
  },
}

export default creator
