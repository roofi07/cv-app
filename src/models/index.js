import creator from './creator'

export default {
  creator,
}
