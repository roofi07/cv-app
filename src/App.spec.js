import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import { init } from '@rematch/core'

import App from './App'
import models from './models'

const store = init({
  models,
})

describe('App Component', () => {
  describe('rendering', () => {
    it('should render as expected', () => {
      const wrapper = shallow(<Provider store={store}><App /></Provider>)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
