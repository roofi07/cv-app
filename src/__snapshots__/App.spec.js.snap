// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component rendering should render as expected 1`] = `<App />`;
